streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -628,15 +647,27 @@ public void onAssignment(final Assignment assignment) {
         Collections.sort(partitions, PARTITION_COMPARATOR);
 
         final AssignmentInfo info = AssignmentInfo.decode(assignment.userData());
-        final int usedVersion = info.version();
+        final int receivedAssignmentMetadataVersion = info.version();
+
+        if (usedSubscriptionMetadataVersion > receivedAssignmentMetadataVersion) {
+            if (receivedAssignmentMetadataVersion >= 3) {
```

Comment:
I think, this cannot really happen because it would require the leader to be older than this member of the consumer group -- and thus, the leader would have failed before if it does not support version probing yet (because the sent subscription metadata is newer than what the leader understands) and there would not be a back call to `onAssignment()`